examples/test_surf_diff.cpp

```diff
@@ -0,0 +1,334 @@
+#include "mfem.hpp"
+#include <fstream>
+#include <iostream>
+
+using namespace std;
+using namespace mfem;
+
+Mesh * GetMesh(int type);
+
+void trans(const Vector &x, Vector &r);
+
+void sigmaFunc(const Vector &x, DenseMatrix &s);
+
+double uExact(const Vector &x)
+{
+   return (0.25 * (2.0 + x[0]) - x[2]) * (x[2] + 0.25 * (2.0 + x[0]));
+}
+
+int main(int argc, char *argv[])
+{
+   // 1. Parse command-line options.
+   int order = 3;
+   int mesh_type = 4; // Default to Quadrilateral mesh
+   int ref_levels = 0;
+   bool static_cond = false;
+   bool pa = false;
+   const char *device_config = "cpu";
+   bool visualization = true;
+
+   OptionsParser args(argc, argv);
+   args.AddOption(&mesh_type, "-mt", "--mesh-type",
+                  "Mesh type: 3 - Triangular, 4 - Quadrilateral.");
+   args.AddOption(&ref_levels, "-r", "--refine",
+                  "Number of times to refine the mesh uniformly in serial.");
+   args.AddOption(&order, "-o", "--order",
+                  "Finite element order (polynomial degree) or -1 for"
+                  " isoparametric space.");
+   args.AddOption(&static_cond, "-sc", "--static-condensation", "-no-sc",
+                  "--no-static-condensation", "Enable static condensation.");
+   args.AddOption(&pa, "-pa", "--partial-assembly", "-no-pa",
```

Comment:
`-pa` and `-device` options would not work for this case. Is this example staying here or only the unit test would stay?

Reply:
Good point. I have removed the `pa` and device related code.

I don't feel strongly about this example either way. It might be good to keep it if we don't have another example of solving PDEs on embedded surfaces. Let's take a vote. By a show of thumbs up or down (@sshiraiwa, @psocratis, @tzanio), should we make `test_surf_diff.cpp` an example?